### ISSUE TITLE
lada-cache:flush generates Redis-Cluster error - cannot use "KEYS"

### ISSUE THREAD

**Issue by madsem** [spiritix/lada-cache#62]
Trying to run lada-cache:flush before a new migration (and also trying to run the db:seed command)
generates a redis error:

```bash
panel.vm $ php artisan lada-cache:flush

   Predis\NotSupportedException  : Cannot use 'KEYS' with redis-cluster.
```

This is my database.php config:

```php
'redis' => [

        'client' => 'predis',

        'options' => [
            'cluster' => 'redis',
        ],

        'clusters' => [
            'default' => [
                'host' => env('REDIS_HOST', 'localhost'),
                'password' => env('REDIS_PASSWORD', null),
                'port' => env('REDIS_PORT', 6379),
                'database' => 0,
            ],
        ],

    ],
```

**Comment by spiritix** [comment]
Didn't test the flush command so far with a cluster, to be honest. Predis does currently [not support](https://github.com/nrk/predis/issues/172) several commands for cluster environments, will need to develop a workaround for this. 

Are you actually using a cluster?

**Comment by madsem** [comment]
yeah using a cluster in production on AWS Elasticache. A workaround is to change the config locally to default when running a migration, but that just introduces problems because I can't be sure if it will work the same way in production I guess.

**Comment by spiritix** [comment]
Well, what you always can do is flushing the entire Redis DB on production instead of just Lada Cache. That will certainly work if your application doesn't depend on any data in Redis being persistent.

**Comment by madsem** [comment]
Ok true, or I switch to cluster mode disabled. Luckily I am flexible there.
Laravel itself also has issues with Cache::tags() & cluster mode enabled.

**Comment by madsem** [comment]
This is the fix suggested by someone in the Laravel repo for using flush with cache tags, it might also solve this issue here as I saw Lada cache also isn't using curly braces:
https://github.com/laravel/framework/pull/17792/files

https://github.com/laravel/framework/issues/17713#issuecomment-278940512

**Comment by spiritix** [comment]
I am not sure if that's going to work, according to the author of Predis one would need to iterate over all connections: https://github.com/nrk/predis/issues/338#issuecomment-222321432

It appears a bit strange to me why this should be required since for all other commands iterating is not required, but ok, either this or your proposed solution should work for sure.

**Comment by madsem** [comment]
Can't say anything about the connection looping, I tested the curly brace solution with Laravel 5.6 and there it seemed to fix the issue indeed, after implementation the errors about 'Cannot use 'DEL' with redis-cluster.' in combination with Cache::tags() were gone.

**Comment by xwiz** [comment]
@spiritix What's the state of this issue currently?

**Comment by xwiz** [comment]
Thanks @spiritix 

Looks like that will happens as I'm running into some issues. Good to know you're open to contributions.

**Comment by xwiz** [comment]
I think for contributors it would be great to have documented the current conditions that trigger the flush cache automatically. By default I assume that would be created, updated, and deleted but one needs to delve into the code to confirm.

**Comment by spiritix** [comment]
So the flush command is **never** triggered automatically. It's only for debug and maintenance purposes, that's why I am not scheduling this with high priority at the moment. The invalidation of the cached items is not related to flushing the cache, you may have a look at the Wiki to see how this works (https://github.com/spiritix/lada-cache/wiki).
